src/lazy/value.rs

```diff
@@ -729,27 +782,109 @@ mod tests {
         Ok(())
     }
 
-    #[ignore] // https://github.com/amazon-ion/ion-rust/issues/951
-    #[test]
-    fn row_and_column_should_not_be_present_for_binary_ion() -> IonResult<()> {
-        let ion_bytes = [
-            0xE0u8, 0x01, 0x00, 0xEA, // IVM
-            0x85, 65, 10, 66, 10, 67, // String: "A\nB\nC"
-            0x85, 68, 10, 69, 10, 70, // String: "D\nE\nF"
-        ];
-        let values: Vec<_> =
-            Reader::new(v1_0::Binary, ion_bytes.as_slice())?.collect::<IonResult<_>>()?;
+    #[rstest]
+    #[case::no_crlf( "{}\"hello\"",       [(1, 1), (1, 3)])]
+    #[case::lf_lf_lf("{}\n\n\n\"hello\"", [(1, 1), (4, 1)] )]
+    #[case::lf_lf_cr("{}\n\n\r\"hello\"", [(1, 1), (4, 1)] )]
+    #[case::lf_cr_cr("{}\n\r\r\"hello\"", [(1, 1), (4, 1)] )]
+    // FIXME: Currently failing because of https://github.com/amazon-ion/ion-rust/issues/955
+    // #[case::lf_cr_lf("{}\n\r\n\"hello\"", [(1, 1), (3, 1)] )]
+    // #[case::cr_lf_lf("{}\r\n\n\"hello\"", [(1, 1), (3, 1)] )]
+    // #[case::cr_lf_cr("{}\r\n\r\"hello\"", [(1, 1), (3, 1)] )]
+    // #[case::cr_cr_lf("{}\r\r\n\"hello\"", [(1, 1), (3, 1)] )]
+    #[case::cr_cr_cr("{}\r\r\r\"hello\"", [(1, 1), (4, 1)] )]
```

Comment:
Making a note of our offline conversation: I was surprised to see that we support standalone `\r`s in our row/column calculations as these are only (?) used in pre-2000-era mac systems. Counting `\n`s (which inherently includes `\r\n`s) should be a `memchr` SIMD operation while counting `\r`, `\n`, and `\r\n`s separately requires a state machine.